### ISSUE TITLE
loss function 

### ISSUE THREAD

**Issue by ssetty** [matterport/Mask_RCNN#1111]
During custom dataset training I observe below losses:
n_mask_loss, rpn_class_loss,rpn_bbox_loss,mrcnn_class_loss,mrcnn_bbox_loss,mrcnn_mask_loss,val_loss, , val_rpn_class_loss, val_rpn_bbox_loss,val_mrcnn_class_loss,val_mrcnn_bbox_loss,val_mrcnn_mask_loss.

What is significance and meaning of each loss and how is it minimized?

**Comment by stygian2a** [comment]
So, you have 5 "small" losses:

    rpn_class_loss : How well the Region Proposal Network separates background with objetcs
    rpn_bbox_loss : How well the RPN localize objects
    mrcnn_bbox_loss : How well the Mask RCNN localize objects
    mrcnn_class_loss : How well the Mask RCNN recognize each class of object
    mrcnn_mask_loss : How well the Mask RCNN segment objects

That makes a bigger loss:

    loss : A combination (surely an addition) of all the smaller losses.

All of those losses are calculated on the training dataset.

The losses for the validation dataset are those starting with 'val'

It is minimized during the training of the network. You can see more details of the loss functions in the original article : https://arxiv.org/pdf/1703.06870.pdf

Hope this helps.

**Comment by Eldad27** [comment]
@ssetty @stygian2a Please which folders do I locate the code implementation for these losses? Can you please help?

**Comment by SilasGK** [comment]
@Eldad27 The loss functions are defined in the mrcnn/model.py, starting from line 1040.
